src/tests/unit/platform/test_policy_ruleconfigs.py

```diff
@@ -280,3 +289,276 @@ def on_delete(url, body):
+    assert groups[1].description == "No playing DOOM!"
+    rules = groups[1].rules_
+    assert len(rules) == 1
+    assert rules[0].name == "DoomyDoomsOfDoom"
+
+
+def test_modify_remove_rule_from_host_based_firewall(cbcsdk_mock):
+    """Tests modifying a host-based firewall rule configuration by removing a rule."""
+    put_called = False
+
+    def on_put(url, body, **kwargs):
+        nonlocal put_called
+        assert body == HBFW_REMOVE_RULE_PUT_REQUEST
+        put_called = True
+        return copy.deepcopy(HBFW_REMOVE_RULE_PUT_RESPONSE)
+
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/1492/configs/presentation',
+                             POLICY_CONFIG_PRESENTATION)
+    cbcsdk_mock.mock_request('PUT', '/policyservice/v1/orgs/test/policies/1492/rule_configs/host_based_firewall',
+                             on_put)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 1492, copy.deepcopy(FULL_POLICY_5), False, True)
+    assert not policy.is_dirty()
+    rule_config = policy.host_based_firewall_rule_config
+    result_groups = [group for group in rule_config.rule_groups if group.name == "Crapco_firewall"]
+    assert len(result_groups) == 1
+    result_rules = [rule for rule in result_groups[0].rules_ if rule.name == "DoomyDoomsOfDoom"]
+    assert len(result_rules) == 1
+    result_rules[0].remove()
+    assert policy.is_dirty()
+    rule_config.save()
+    assert put_called
+    result_groups = [group for group in rule_config.rule_groups if group.name == "Crapco_firewall"]
+    assert len(result_groups) == 1
+    result_rules = result_groups[0].rules_
+    assert len(result_rules) == 1
+    assert result_rules[0].name == "my_first_rule"
+
+
+def test_modify_remove_rule_group_from_host_based_firewall(cbcsdk_mock):
+    """Tests modifying a host-based firewall rule configuration by removing a rule group."""
+    put_called = False
+
+    def on_put(url, body, **kwargs):
+        nonlocal put_called
+        assert body == HBFW_REMOVE_RULE_GROUP_PUT_REQUEST
+        put_called = True
+        return copy.deepcopy(HBFW_REMOVE_RULE_GROUP_PUT_RESPONSE)
+
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/1492/configs/presentation',
+                             POLICY_CONFIG_PRESENTATION)
+    cbcsdk_mock.mock_request('PUT', '/policyservice/v1/orgs/test/policies/1492/rule_configs/host_based_firewall',
+                             on_put)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 1492, copy.deepcopy(FULL_POLICY_5), False, True)
+    assert not policy.is_dirty()
+    rule_config = policy.host_based_firewall_rule_config
+    result_groups = [group for group in rule_config.rule_groups if group.name == "Isolate"]
+    assert len(result_groups) == 1
+    result_groups[0].remove()
+    assert policy.is_dirty()
+    rule_config.save()
+    assert put_called
+    result_groups = rule_config.rule_groups
+    assert len(result_groups) == 1
```

Comment:
Why does the length of the group remain 1 if it was removed above? 

Also can we add a test which has multiple rules in a rule group to ensure the correct rule is removed as only 1 rulegroup or rule doesn't test the new add/remove logic

Reply:
> Why does the length of the group remain 1 if it was removed above?

We're measuring two different things. Lines 556-557 are measuring the number of groups which were found matching the name "Isolate." That should always be 1.  Lines 562-563 are measuring the number of groups left _after_ the "Isolate" group is removed. That should also be 1.

> Also can we add a test which has multiple rules in a rule group to ensure the correct rule is removed as only 1 rulegroup or rule doesn't test the new add/remove logic

I thought of this. The definition of the policy in the mock `FULL_POLICY_5` has two rule groups, one of which has two rules, the other of which has only one.  It was added for the specific purpose of testing rule addition and removal.  The structure of the rule groups in `FULL_POLICY_5` is as follows:
* Group `Crapco_firewall`
  * Rule `my_first_rule`
  * Rule `DoomyDoomsOfDoom`
* Group `Isolate`
  * Rule `BlockIRC`